Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/CalculateMonteCarloAbsorption.py

```diff
@@ -0,0 +1,670 @@
+                        transpose_alg = self.createChildAlgorithm("Transpose", enableLogging=False)
+                        transpose_alg.setProperty("InputWorkspace", workspace)
+                        transpose_alg.execute()
+                        workspace = transpose_alg.getProperty("OutputWorkspace").value
+                    return self._create_waves_indirect_elastic(workspace)
+                else:
+                    convert_unit_alg.setProperty("EFixed", self._efixed)
+
+            convert_unit_alg.setProperty("InputWorkspace", workspace)
+            convert_unit_alg.setProperty("Target", 'Wavelength')
+            convert_unit_alg.setProperty("EMode", self._emode)
+
+            convert_unit_alg.execute()
+
+            return convert_unit_alg.getProperty("OutputWorkspace").value
+
+    def _convert_from_wavelength(self, workspace):
+
+        convert_unit_alg = self.createChildAlgorithm("ConvertUnits", enableLogging=False)
+        convert_unit_alg.setProperty("Target", self._sample_unit)
+
+        if self._sample_unit != 'Wavelength':
+
+            if self._emode == 'Indirect':
+
+                if self._indirect_q_axis is not None:
+
+                    if self._indirect_q_axis == 'X':
+                        transpose_alg = self.createChildAlgorithm("Transpose", enableLogging=False)
+                        transpose_alg.setProperty("InputWorkspace", workspace)
+                        transpose_alg.execute()
+                        workspace = transpose_alg.getProperty("OutputWorkspace").value
+                        workspace.setX(0, self._q_values)
+                        workspace.getAxis(0).setUnit("MomentumTransfer")
+                        return workspace
+                    convert_unit_alg.setProperty("Target", "MomentumTransfer")
+                convert_unit_alg.setProperty("EFixed", self._efixed)
+
+            convert_unit_alg.setProperty("InputWorkspace", workspace)
+            convert_unit_alg.setProperty("EMode", self._emode)
+            convert_unit_alg.execute()
+            return convert_unit_alg.getProperty("OutputWorkspace").value
+
+        else:
+            return workspace
+
+    # ------------------------------- Converting IndirectElastic to wavelength ------------------------------
+
+    def _create_waves_indirect_elastic(self, workspace):
+        """
+        Creates a wavelength workspace, from the workspace with the specified input workspace
+        name, using an Elastic instrument definition file. E-Mode must be Indirect and the y-axis
+        of the input workspace must be in units of Q.
+
+        :param workspace:   The input workspace.
+        :return:            The output wavelength workspace.
+        """
+        self._q_values = workspace.getAxis(1).extractValues()
+
+        # ---------- Load Elastic Instrument Definition File ----------
+
+        idf_name = workspace.getInstrument().getName() + '_elastic_Definition.xml'
```

Comment:
This part seems ISIS-bound. We do not have any `elastic_Definition` for ILL instruments. But this algorithm should run as well. Can you explain what it is, and why do you need a special IDF for indirect elastic?